File: sol/solution/0001-0100/0005. Longest Palindromic Substring/Solution.scala
object Solution {
    def longestPalindrome(s: String): String = {
        val n = s.length
        var start = 0
        var maxLen = 1
        for (i <- 0 until n) {
            var left = i
            var right = i
            while (left >= 0 && right < n && s(left) == s(right)) {
                if (right - left + 1 > maxLen) {
                    start = left
                    maxLen = right - left + 1
                }
                left -= 1
                right += 1
            }
            left = i
            right = i + 1
            while (left >= 0 && right < n && s(left) == s(right)) {
                if (right - left + 1 > maxLen) {
                    start = left
                    maxLen = right - left + 1
                }
                left -= 1
                right += 1
            }
        }
        s.substring(start, start + maxLen)
    }
}

File: sol/solution/0001-0100/0005. Longest Palindromic Substring/Solution.java
class Solution {
    public String longestPalindrome(String s) {
        int n = s.length();
        int start = 0, maxLen = 1;
        for (int i = 0; i < n; i++) {
            int left = i, right = i;
            while (left >= 0 && right < n && s.charAt(left) == s.charAt(right)) {
                if (right - left + 1 > maxLen) {
                    start = left;
                    maxLen = right - left + 1;
                }
                left--;
                right++;
            }
            left = i; right = i + 1;
            while (left >= 0 && right < n && s.charAt(left) == s.charAt(right)) {
                if (right - left + 1 > maxLen) {
                    start = left;
                    maxLen = right - left + 1;
                }
                left--;
                right++;
            }
        }
        return s.substring(start, start + maxLen);
    }
}